main.go

```diff
@@ -1,143 +1,129 @@
-
-	cmd.Flags().StringVarP(&username,
-		"username",
-		"u",
-		os.Getenv("INFLUXDB_USER"),
-		"the username for the given db, defaults to value of INFLUXDB_USER env variable")
-
-	cmd.Flags().StringVarP(&password,
-		"password",
-		"p",
-		os.Getenv("INFLUXDB_PASS"),
-		"the password for the given db, defaults to value of INFLUXDB_PASS env variable")
-
-	/* Non-sensitive arguments */
-	cmd.Flags().StringVarP(&dbName,
-		"db-name",
-		"d",
-		"",
-		"the influxdb to send metrics to")
-
-	cmd.Flags().StringVarP(&precision,
-		"precision",
-		"",
-		"s",
-		"the precision value of the metric")
-
-	cmd.Flags().BoolVarP(&insecureSkipVerify,
-		"insecure-skip-verify",
-		"i",
-		false,
-		"if true, the influx client skips https certificate verification")
-
-	_ = cmd.MarkFlagRequired("db-name")
-
-	return cmd
+type HandlerConfig struct {
+	sensu.PluginConfig
+	Addr               string
+	Username           string
+	Password           string
+	DbName             string
+	Precision          string
+	InsecureSkipVerify bool
 }
 
-func run(cmd *cobra.Command, args []string) error {
-	if len(args) != 0 {
-		_ = cmd.Help()
-		return fmt.Errorf("invalid argument(s) received")
-	}
-
-	if addr == "" {
-		addr = "http://localhost:8086"
-	}
+const (
+	addr               = "addr"
+	username           = "username"
+	password           = "password"
+	dbName             = "db-name"
+	precision          = "precision"
+	insecureSkipVerify = "insecure-skip-verify"
+)
 
-	if stdin == nil {
-		stdin = os.Stdin
+var (
+	config = HandlerConfig{
+		PluginConfig: sensu.PluginConfig{
+			Name:  "sensu-influxdb-handler",
+			Short: "an influxdb handler built for use with sensu",
+		},
 	}
 
-	eventJSON, err := ioutil.ReadAll(stdin)
-	if err != nil {
-		return fmt.Errorf("failed to read stdin: %s", err)
+	influxdbConfigOptions = []*sensu.PluginConfigOption{
+		{
+			Path:      addr,
+			Env:       "INFLUXDB_ADDR",
+			Argument:  addr,
+			Shorthand: "a",
+			Default:   "http://localhost:8086",
+			Usage:     "the address of the influxdb server, should be of the form 'http://host:port', defaults to 'http://localhost:8086' or value of INFLUXDB_ADDR env variable",
+			Value:     &config.Addr,
+		},
+		{
+			Path:      username,
+			Env:       "INFLUXDB_USER",
+			Argument:  username,
+			Shorthand: "u",
+			Default:   "",
+			Usage:     "the username for the given db, defaults to value of INFLUXDB_USER env variable",
+			Value:     &config.Username,
+		},
+		{
+			Path:      password,
+			Env:       "INFLUXDB_PASS",
+			Argument:  password,
+			Shorthand: "p",
+			Default:   "",
+			Usage:     "the password for the given db, defaults to value of INFLUXDB_PASS env variable",
+			Value:     &config.Password,
+		},
+		{
+			Path:      dbName,
+			Argument:  dbName,
+			Shorthand: "d",
+			Default:   "",
+			Usage:     "the influxdb to send metrics to",
+			Value:     &config.DbName,
+		},
+		{
+			Path:      precision,
+			Argument:  precision,
+			Shorthand: "s",
```

Comment:
There is no shorthand for the precision field.
```suggestion
			Shorthand: "",
```